nixops/state/state_helper.py

```diff
@@ -0,0 +1,48 @@
+import json
+import collections
+import nixops.util
+
+
+def _subclasses(cls):
+    sub = cls.__subclasses__()
+    return [cls] if not sub else [g for s in sub for g in _subclasses(s)]
+
+
+class StateDict(collections.MutableMapping):
+    """
+       An implementation of a MutableMapping container providing
+       a python dict like behavior for the NixOps state file.
+    """
+    # TODO implement __repr__ for convenience e.g debuging the structure
+    def __init__(self, depl, id):
+        super(StateDict, self).__init__()
+        self._state = depl._state
+        self.uuid = depl.uuid
+        self.id = id
+
+    def __setitem__(self, key, value):
+        self._state.set_resource_attrs(self.uuid, self.id, {key:value})
+
+    def __getitem__(self, key):
+        value = self._state.get_resource_attr(self.uuid, self.id, key)
```

Comment:
I get an error here https://pastebin.com/N8E90uke because of the inconsistency between `value` type:
- sqlite always returns a string (though this string may contain json, number or boolean)
- json state file returns already parsed value

Reply:
fix https://github.com/NixOS/nixops/pull/969/commits/e1b9158f67dc7673e9d84eaf9e3a5b73c21cb31a